### PR TITLE
[CNFT2-1577] - Search Question

### PR DESCRIPTION
**Pull request by lramakr**
## Description

    Developed Search Question UI

## Tickets

* [Jira Ticket] https://cdc-nbs.atlassian.net/browse/CNFT2-1577

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/components/Breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    };
+
+    return (
+        <div className={`pages-breadcrumb ${className || ''}`} onClick={goBack}>
+            <Icon.ArrowBack size={3} />
+            <h4>{header}</h4>
+            {currentPage ? <h5>{currentPage}</h5> : null}
+        </div>
+    );
+};
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/components/Breadcrumb/Breadcrumb.tsx:26]: Issue: The `Breadcrumb` component was moved to `apps/modernization-ui/src/breadcrumb/Breadcrumb.tsx`. When merging main a duplicate `Breadcrumb` component was added. 

Please delete this `Breadcrumb` component and use the shared one.

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx [hunk 2]
```diff
@@ -2,12 +2,16 @@ import { PagesTab } from 'apps/page-builder/generated';
...
 type Props = {
     tab: PagesTab;
 };
 export const PageContent = ({ tab }: Props) => {
+    const navigateTo = useNavigate();
+    const { pageId } = useParams();
     const handleAddQuestion = (subsection: number) => {
+        navigateTo(`/page-builder/pages/${pageId}/question-library/${subsection}`);
         console.log('add question not yet implemented', subsection);
     };
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx:16]: Question: Isn't the add question flow supposed to display within a modal?

File: apps/modernization-ui/src/apps/page-builder/pages/PageBuilder/PageBuilder.tsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { PageBuilderSideNav } from '../../components/Navigation/PageBuilderSideN
 import './PageBuilder.scss';
 
 type Props = {
+    page?: string;
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/pages/PageBuilder/PageBuilder.tsx:5]: Question: Is this needed? the `PageBuilder.tsx` is no longer used in the new `Edit` page so there should be no need to make changes to it.

File: apps/modernization-ui/src/apps/page-builder/pages/QuestionLibrary/QuestionLibraryTable.tsx [hunk 4]
```diff
@@ -125,42 +128,38 @@ export const QuestionLibraryTable = ({ summaries, pages, qtnModalRef }: Props) =
     };
 
     const handleAddQsntoPage = async () => {
+        console.log('handleAddQsntoPage...');
         // TODO need to add logic for find orderNumber and Id
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/pages/QuestionLibrary/QuestionLibraryTable.tsx:132]: Issue: `console.log` can be removed

File: apps/modernization-ui/src/routes/AppRoutes.tsx [hunk 5]
```diff
@@ -64,6 +65,7 @@ export const AppRoutes = () => {
                 <Route path=":pageId">
                     <Route index element={<PreviewPage />} />
                     <Route path="edit" element={<Edit />} />
+                    <Route path="question-library/:subsectionId" element={<QuestionLibrary />} />
```
Comment (mpeels) [inline apps/modernization-ui/src/routes/AppRoutes.tsx:68]: As mentioned above, I don't believe the `QuestionLibrary` appears outside of a modal within the context of the `Edit` page flow. I could be wrong.

**Review by mpeels — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx (hunk 2 above)
Comment (lramakr) [reply apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx:16]: I have removed the model and displaying with in the page itself.

File: apps/modernization-ui/src/routes/AppRoutes.tsx (hunk 5 above)
Comment (lramakr) [reply apps/modernization-ui/src/routes/AppRoutes.tsx:68]: It will be with in the context. but it is not opening in a model

Comment (lramakr) [reply apps/modernization-ui/src/routes/AppRoutes.tsx:68]: it should open in a model. So working on it

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx (hunk 2 above)
Comment (lramakr) [reply apps/modernization-ui/src/apps/page-builder/page/management/edit/content/PageContent.tsx]: Fixed to open in the model

File: apps/modernization-ui/src/routes/AppRoutes.tsx (hunk 5 above)
Comment (lramakr) [reply apps/modernization-ui/src/routes/AppRoutes.tsx]: Fixed to open in the model

**Review by mpeels — APPROVED** [review]

**Pull request merged**